presto-docs/src/main/sphinx/language/types.rst

```diff
@@ -169,22 +172,62 @@ See also :doc:`/language/timestamp`
 
     Example: ``TIME '01:02:03.456 America/Los_Angeles'``
 
+.. _timestamp-data-type:
+
 ``TIMESTAMP``
 ^^^^^^^^^^^^^
 
     Instant in time that includes the date and time of day without a time zone.
     Values of this type are parsed and rendered in the session time zone.
+    Equivalent to ``TIMESTAMP WITHOUT TIME ZONE``, ``TIMESTAMP(3)``, and
+    ``TIMESTAMP(3) WITHOUT TIME ZONE``.
 
     Example: ``TIMESTAMP '2001-08-22 03:04:05.321'``
 
+``TIMESTAMP(P)``
+^^^^^^^^^^^^^^^^
+
+    Instant in time that includes the date and time of day without a time zone
+    with ``P`` digits of precision for the fraction of seconds.
+
+    The default ``TIMESTAMP`` is equivalent to ``TIMESTAMP(3)`` and means that
+    up millisecond values are included. A precision of up to 12, so picoseconds,
+    is supported.
+
+    Casting to lower precision causes rounding, and to higher precision causes
+    appended zero values.
+
+    The following examples illustrate the behavior::
+
+        SELECT TIMESTAMP('2020-06-10 15:55:23.383345');
+        -- 2020-06-10 15:55:23.383345
+
+        SELECT typeof(TIMESTAMP '2020-06-10 15:55:23.383345');
+        -- timestamp(6)
+
+        SELECT cast(TIMESTAMP '2020-06-10 15:55:23.383345' as TIMESTAMP(1));
+         -- 2020-06-10 15:55:23.4
+
+        SELECT cast(TIMESTAMP '2020-06-10 15:55:23.383345' as TIMESTAMP(12));
+        -- 2020-06-10 15:55:23.383345000000
+
 ``TIMESTAMP WITH TIME ZONE``
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
     Instant in time that includes the date and time of day with a time zone.
     Values of this type are rendered using the time zone from the value.
+    Equivalent to ``TIMESTAMP(3) WITH TIME ZONE``.
 
     Example: ``TIMESTAMP '2001-08-22 03:04:05.321 America/Los_Angeles'``
 
+``TIMESTAMP(P) WITH TIME ZONE``
```

Comment:
This is not yet supported in master. It's in progress: https://github.com/prestosql/presto/pull/3947

Reply:
oops ... let me take it back out..

Reply:
done